openshift/openshift.go

```diff
@@ -153,19 +153,17 @@ func (c *openshiftClient) convertDockerImageReference(ref string) (string, error
 	if len(parts) != 2 {
 		return "", fmt.Errorf("Invalid format of docker reference %s: missing '/'", ref)
 	}
-	// Sanity check that the reference is at least plausibly similar, i.e. uses the hard-coded port we expect.
-	if !strings.HasSuffix(parts[0], ":5000") {
-		return "", fmt.Errorf("Invalid format of docker reference %s: expecting port 5000", ref)
-	}
 	return c.dockerRegistryHostPart() + "/" + parts[1], nil
 }
 
 // dockerRegistryHostPart returns the host:port of the embedded Docker Registry API endpoint
-// FIXME: There seems to be no way to discover the correct:host port using the API, so hard-code our knowledge
-// about how the OpenShift Atomic Registry is configured, per examples/atomic-registry/run.sh:
-// -p OPENSHIFT_OAUTH_PROVIDER_URL=https://${INSTALL_HOST}:8443,COCKPIT_KUBE_URL=https://${INSTALL_HOST},REGISTRY_HOST=${INSTALL_HOST}:5000
 func (c *openshiftClient) dockerRegistryHostPart() string {
-	return strings.SplitN(c.ref.baseURL.Host, ":", 2)[0] + ":5000"
+	registryPort := strings.SplitN(c.ref.registry, ":", 2)[1]
+	if registryPort == "" {
+		// FIXME(runcom): should we error out @mitr? can this happen at all?
+		registryPort = "5000"
```

Comment:
This should behave exactly the same as if a non-OpenShift docker reference does not specify a port; I.e., I think, leave the port part empty (and let `dockerClient` use whatever default it uses).